### PR TITLE
feat: Add ability to edit personal and household information on the user profile

### LINKED ISSUE #137: feat: Add ability to edit personal and household information on the user profile
there are buttons to edit these fields, but they currently only pull up empty bottom drawers. Add the ability to edit this information and save it to the database

**Don't** allow them to update their emails. Name/phone number should be editable in the user info section however

In "edit household" mode -- allow them to edit address, pets, accessibility needs, and notes. Display the household members but don't allow edits

~~For household members, don't allow updates to `Person`s that have `User` profiles.~~ (this whole thing is out of scope for this task, pls disregard)

### PR DESCRIPTION
**Pull request by hinxcode**
This pull request introduces several new features and improvements to the `support_sphere` project, focusing on enhancing user management and profile handling. The most significant changes include adding new methods for updating user and household information, introducing a new `ProfileSection` component, and integrating these updates into the profile page.

### New Features and Enhancements:

* **User and Household Updates:**
  * Added `updateUserPhoneNumber` method to `AuthenticationRepository` for updating user phone numbers. (`src/support_sphere/lib/data/repositories/authentication.dart`)
  * Introduced methods in `UserRepository` for updating user names and household details. (`src/support_sphere/lib/data/repositories/user.dart`)
  * Implemented corresponding update methods in `UserService` for updating person and household details in the database. (`src/support_sphere/lib/data/services/user_service.dart`)

* **Profile Section Component:**
  * Created a new `ProfileSection` component to modularize profile sections with editable modals. (`src/support_sphere/lib/presentation/components/profile_section.dart`)

### Integration into Profile Page:

* **Profile Page Updates:**
  * Replaced the old `_ProfileSection` with the new `ProfileSection` component in the profile body. (`src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart`) [[1]](diffhunk://#diff-4f8fa883208165c32b6369633dd5323a173c40e56237aad060b3115491324856L87-R99) [[2]](diffhunk://#diff-4f8fa883208165c32b6369633dd5323a173c40e56237aad060b3115491324856R197-R206)
  * Added form builders for editing personal and household information within the `ProfileSection` modals. (`src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart`) [[1]](diffhunk://#diff-4f8fa883208165c32b6369633dd5323a173c40e56237aad060b3115491324856R135-R185) [[2]](diffhunk://#diff-4f8fa883208165c32b6369633dd5323a173c40e56237aad060b3115491324856L258-R256)

### Additional Changes:

* **Error Handling:**
  * Added a new error message string for cases where no user is signed in. (`src/support_sphere/lib/constants/string_catalog.dart`)
  * Implemented error handling in `AuthService` for user phone number updates. (`src/support_sphere/lib/data/services/auth_service.dart`)

These changes collectively improve the user experience by enabling easier updates to user profiles and household information, while also enhancing the modularity and maintainability of the codebase.

### Related issue

Resolves #137 

### DEMO

image
image
image
image

### CODE REVIEW

File: src/support_sphere/pubspec.lock [hunk 1]
```diff
@@ -1049,4 +1057,4 @@ packages:
     version: "2.0.2"
 sdks:
   dart: ">=3.5.0 <4.0.0"
-  flutter: ">=3.22.0"
+  flutter: ">=3.24.0"
```
Comment (hinxcode) [inline src/support_sphere/pubspec.lock:1060]: flutter had been upgraded when I installed flutter_form_builder but I can take it back if that makes more sense to everyone.

File: src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart [hunk 2]
```diff
@@ -189,6 +132,57 @@ class _PersonalInformation extends StatelessWidget {
...
             ),
           ],
+          modalBody: FormBuilder(
+            key: formKey,
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                FormBuilderTextField(
+                  name: 'givenName',
+                  decoration: const InputDecoration(labelText: 'Given Name'),
```
Comment (lsetiawan) [inline src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart:142]: Can you put these label texts into the string catalog constants? We'll need for all the hardcoded label text

File: src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart [hunk 3]
```diff
@@ -200,14 +194,16 @@ class _HouseholdInformation extends StatelessWidget {
...
+
     return BlocBuilder<ProfileCubit, ProfileState>(
       buildWhen: (previous, current) => previous.household != current.household,
       builder: (context, state) {
         Household? household = state.household;
         String address = household?.address ?? '';
         String pets = household?.pets ?? '';
         String notes = household?.notes ?? '';
-        String accessibilityNeeds = household?.accessibility_needs ?? 'None';
+        String accessibilityNeeds = household?.accessibility_needs ?? '';
```
Comment (lsetiawan) [inline src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart:206]:
```suggestion
        String accessibilityNeeds = household?.accessibility_needs ?? 'Not Applicable';
```

I think it makes more sense for a person without any accessibility needs to be Not Applicable.. probably want to put that default string into the strings catalog.

File: src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart [hunk 4]
```diff
@@ -255,7 +253,7 @@ class _HouseholdInformation extends StatelessWidget {
               mainAxisAlignment: MainAxisAlignment.spaceBetween,
               children: [
                 Text("Accessiblity Needs"),
-                Text(accessibilityNeeds),
+                Text(accessibilityNeeds.isEmpty ? "None" : accessibilityNeeds),
```
Comment (lsetiawan) [inline src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart:256]:
```suggestion
                Text(accessibilityNeeds.isEmpty ? "Not Applicable" : accessibilityNeeds),
```

File: src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart [hunk 5]
```diff
@@ -276,6 +274,63 @@ class _HouseholdInformation extends StatelessWidget {
...
             )
           ],
+          modalBody: FormBuilder(
+            key: formKey,
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                FormBuilderTextField(
+                  name: 'address',
+                  decoration: const InputDecoration(labelText: 'Address'),
```
Comment (lsetiawan) [inline src/support_sphere/lib/presentation/pages/main_app/profile/profile_body.dart:284]: Let's put the label text to a string catalog here.

**Pull request merged**